mods/pageviews.js

```diff
@@ -365,7 +370,40 @@ PJVS.prototype.pageviewsForTops = function(restbase, req) {
 
     }).catch(notFoundCatcher);
 
-    return dataRequest.then(normalizeResponse);
+    return dataRequest.then(normalizeResponse).then(function(res) {
+        if (res.body.items) {
+            res.body.items.forEach(function(item) {
+                // prefer the articlesJSON column if it's loaded
+                if (item.articlesJSON !== null) {
+                    try {
+                        item.articles = item.articlesJSON.map(function(a) {
+                            a.article = decodeURIComponent(a.article);
```

Comment:
Wouldn't it be better to have the URI decoded before storing it in Cassandra, so that it doesn't need to be decoded on each request?